state/status.go

```diff
@@ -199,48 +199,75 @@ type statusHistoryArgs struct {
 	filter    status.StatusHistoryFilter
 }
 
-func statusHistory(args *statusHistoryArgs) ([]status.StatusInfo, error) {
-	filter := args.filter
-	if err := args.filter.Validate(); err != nil {
-		return nil, errors.Annotate(err, "validating arguments")
-	}
-	statusHistory, closer := args.st.getCollection(statusesHistoryC)
-	defer closer()
-
+func fetchNStatusResults(size int, key string, timeDelta *time.Duration,
+	date *time.Time, col mongo.Collection) ([]historicalStatusDoc, error) {
```

Comment:
We either need a doc string, or better named parameters, eg "fromDate" instead of just "date".
Is there a reason why we don't just pass in a StatusHistoryFilter struct?
ie fetchStatusResults(col mongo.Collection, key string, filter StatusHistoryFilter)